pkg/util/sort_uniq.go

```diff
@@ -0,0 +1,46 @@
+package util
+
+import "sort"
+
+const insertionSortThreshold = 20
+
+// SortUniqInPlace sorts and remove duplicates from elements in place
+// The returned slice is a subslice of elements
+func SortUniqInPlace(elements []string) []string {
+	if len(elements) < 2 {
+		return elements
+	}
+	size := len(elements)
+	if size <= insertionSortThreshold {
+		insertionSort(elements)
+	} else {
+		// this will trigger an alloc because sorts uses interface{} internaly
+		// which confuses the escape analysis
+		sort.Strings(elements)
+	}
+	return uniqSorted(elements)
+}
+
+func insertionSort(elements []string) {
+	for i := 1; i < len(elements); i++ {
+		temp := elements[i]
+		j := i
+		for j > 0 && temp <= elements[j-1] {
+			elements[j] = elements[j-1]
+			j--
+		}
+		elements[j] = temp
+	}
+}
+
+func uniqSorted(elements []string) []string {
```

Comment:
We should maybe add a small comment that it only works on sorted elements. I know that it's already reflected in the name but that would make it clearer?